From 96097520de2b71437bfc000539f1704f438ee89c Mon Sep 17 00:00:00 2001
From: Dean Sanner <dsanner@us.ibm.com>
Date: Fri, 3 Mar 2017 06:27:34 -0600
Subject: [PATCH v1 1001/1001] Revert "Reducing rng pace rate from 2000 -> 300
 for HW403701"

This reverts commit 420ca4a8abaff4aac15614c4dddc0b37f6c239ac.
---
 .../chips/p9/procedures/hwp/nest/p9_rng_init_phase1.C  | 18 +++---------------
 .../xml/attribute_info/chip_ec_attributes.xml          | 17 -----------------
 2 files changed, 3 insertions(+), 32 deletions(-)

diff --git a/src/import/chips/p9/procedures/hwp/nest/p9_rng_init_phase1.C b/src/import/chips/p9/procedures/hwp/nest/p9_rng_init_phase1.C
index 40fb65a..7a87396 100644
--- a/src/import/chips/p9/procedures/hwp/nest/p9_rng_init_phase1.C
+++ b/src/import/chips/p9/procedures/hwp/nest/p9_rng_init_phase1.C
@@ -134,8 +134,6 @@ const bool     NX_RNG_CFG_ADAPTEST_ENABLE                    = true;
 const uint8_t  NX_RNG_CFG_ST2_RESET_PERIOD                   = 0x07;
 // pace rate (2000)
 const uint16_t NX_RNG_CFG_PACE_RATE                          = 0x07d0;
-// pace rate (300) for HW403701
-const uint16_t NX_RNG_CFG_PACE_RATE_HW403701                 = 0x012c;
 
 
 //------------------------------------------------------------------------------
@@ -154,10 +152,8 @@ p9_rng_init_phase1(const fapi2::Target<fapi2::TARGET_TYPE_PROC_CHIP>& i_target)
     fapi2::buffer<uint64_t> l_rng_st3_data;
 
     uint8_t l_dd1 = 0;
-    uint8_t l_HW403701 = 0;
 
     FAPI_TRY( FAPI_ATTR_GET(fapi2::ATTR_CHIP_EC_FEATURE_RNG_ADAPTEST_SETTINGS, i_target, l_dd1) );
-    FAPI_TRY( FAPI_ATTR_GET(fapi2::ATTR_CHIP_EC_FEATURE_HW403701, i_target, l_HW403701) );
 
     // 1. RNG will start running with FIFO write / self tests disabled (enable doesn't gate the osc; it turns off FIFO
     // writes and self test fails); rng_enable = 0.
@@ -272,6 +268,7 @@ p9_rng_init_phase1(const fapi2::Target<fapi2::TARGET_TYPE_PROC_CHIP>& i_target)
              "Error from putScom (NX RNG Self Test Register 3)");
 
     // 4. If RNG is not broken then host boot sets rng_enable =1.
+
     // update RNG Status and Control Register to engage initialization test
     FAPI_TRY(fapi2::getScom(i_target, PU_NX_RNG_CFG, l_rng_cfg_data),
              "Error from getScom (NX RNG Status and Control Register)");
@@ -289,17 +286,8 @@ p9_rng_init_phase1(const fapi2::Target<fapi2::TARGET_TYPE_PROC_CHIP>& i_target)
     (NX_RNG_CFG_ADAPTEST_ENABLE);
     l_rng_cfg_data.insertFromRight<PU_NX_RNG_CFG_ST2_RESET_PERIOD, PU_NX_RNG_CFG_ST2_RESET_PERIOD_LEN>
     (NX_RNG_CFG_ST2_RESET_PERIOD);
-
-    if(l_HW403701 != 0)
-    {
-        l_rng_cfg_data.insertFromRight<PU_NX_RNG_CFG_PACE_RATE, PU_NX_RNG_CFG_PACE_RATE_LEN>
-        (NX_RNG_CFG_PACE_RATE_HW403701);
-    }
-    else
-    {
-        l_rng_cfg_data.insertFromRight<PU_NX_RNG_CFG_PACE_RATE, PU_NX_RNG_CFG_PACE_RATE_LEN>
-        (NX_RNG_CFG_PACE_RATE);
-    }
+    l_rng_cfg_data.insertFromRight<PU_NX_RNG_CFG_PACE_RATE, PU_NX_RNG_CFG_PACE_RATE_LEN>
+    (NX_RNG_CFG_PACE_RATE);
 
     FAPI_TRY(fapi2::putScom(i_target, PU_NX_RNG_CFG, l_rng_cfg_data),
              "Error from putScom (NX RNG Status and Control Register)");
diff --git a/src/import/chips/p9/procedures/xml/attribute_info/chip_ec_attributes.xml b/src/import/chips/p9/procedures/xml/attribute_info/chip_ec_attributes.xml
index e907802..c96913a 100644
--- a/src/import/chips/p9/procedures/xml/attribute_info/chip_ec_attributes.xml
+++ b/src/import/chips/p9/procedures/xml/attribute_info/chip_ec_attributes.xml
@@ -2512,23 +2512,6 @@
   </attribute>
 
   <attribute>
-    <id>ATTR_CHIP_EC_FEATURE_HW403701</id>
-    <targetType>TARGET_TYPE_PROC_CHIP</targetType>
-    <description>
-      Nimbus DD1: Reduce rng pace from 2000->300 to work around grant unfairness
-    </description>
-    <chipEcFeature>
-      <chip>
-        <name>ENUM_ATTR_NAME_NIMBUS</name>
-        <ec>
-          <value>0x20</value>
-          <test>LESS_THAN</test>
-        </ec>
-      </chip>
-    </chipEcFeature>
-  </attribute>
-
-  <attribute>
     <id>ATTR_CHIP_EC_FEATURE_RING_SAVE_MPIPL</id>
     <targetType>TARGET_TYPE_PROC_CHIP</targetType>
     <description>
-- 
1.8.2.2

